### PR TITLE
Vast swathes of icon and path fixes following major removals.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
TODO:
- [x] Convert base icon checks into a unit test.
- [x] Convert debug verb into a 'list unused icon states' verb.
- ~~Remove unused icon states.~~

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Will do the icon cull in another PR.

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 1]
```diff
@@ -521,23 +521,16 @@
...
-				M.Weaken(15)
-
-			if(4 to 5)
-				if(hasvar(M, "glasses"))
-					if(istype(M:glasses, /obj/item/clothing/glasses/sunglasses))
-						continue
-
-				M.flash_eyes()
-				M.Stun(5)
+	for(var/mob/living/carbon/human/M in viewers(world.view, location))
```
Comment (EmperorJon) [inline code/modules/reagents/Chemistry-Recipes.dm]: IDK if I should be particularly standards-picky about us putting in changes like this that are relatively unrelated to the PR as a whole, unless I'm misunderstanding the need for this here.

Comment (EmperorJon) [reply code/modules/reagents/Chemistry-Recipes.dm]: -this is the sunglasses, isn't it.

Comment (MistakeNot4892) [reply code/modules/reagents/Chemistry-Recipes.dm]: It's replacing the typecheck with casting then checking flash protection. Sunglasses actually don't exist currently.

Comment (EmperorJon) [reply code/modules/reagents/Chemistry-Recipes.dm]: Yep, my guess was you were replacing the sunglasses check because they're gone.

**Comment by EmperorJon** [comment]
Changes fix atmospherics being completely broken. (Still approved) 👍

File: code/controllers/subsystems/initialization/materials.dm [hunk 2]
```diff
@@ -40,6 +40,8 @@ SUBSYSTEM_DEF(materials)
 	. = ..()
 
 /datum/controller/subsystem/materials/proc/get_material_by_name(name)
+	if(!name || name == "")
```
Comment (PsiOmegaDelta) [inline code/controllers/subsystems/initialization/materials.dm]: Merely a note:
`!name` includes `name == "" ` due to how BYOND handles falsy values.

File: code/modules/clothing/ears/earrings.dm [hunk 3]
```diff
@@ -0,0 +1,38 @@
+/obj/item/clothing/ears/earring
+	name = "stud earrings"
+	desc = "An earring of some kind."
+	icon_state = "ear_stud"
+	gender = PLURAL
+	var/material/material = MATERIAL_GLASS
+
+/obj/item/clothing/ears/earring/New(var/newloc, var/new_material)
```
Comment (PsiOmegaDelta) [inline code/modules/clothing/ears/earrings.dm]: `/Initialize(...)`

File: code/modules/clothing/ears/earrings.dm [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	..(newloc)
```
Comment (PsiOmegaDelta) [inline code/modules/clothing/ears/earrings.dm]: `. = ..()`

File: code/controllers/subsystems/initialization/materials.dm (hunk 2 above)
Comment (MistakeNot4892) [reply code/controllers/subsystems/initialization/materials.dm]: As discussed on Discord, I'm too lazy to fix this just now. Will tweak it in a future PR.

Comment (MistakeNot4892) [reply code/controllers/subsystems/initialization/materials.dm]: _I lied._

**Pull request merged**